app/src/main/kotlin/at/bitfire/davdroid/settings/AccountSettingsMigrations.kt

```diff
@@ -58,9 +67,38 @@ class AccountSettingsMigrations @AssistedInject constructor(
         fun create(account: Account, accountSettings: AccountSettings): AccountSettingsMigrations
     }
 
-
     val accountManager: AccountManager = AccountManager.get(context)
 
+    /**
+     * With DAVx5 4.3.3 address book accounts now contain the collection ID as a unique identifier.
+     * We need to update the address book account names.
+     */
+    @Suppress("unused","FunctionName")
+    fun update_16_17() {
+        try {
+            context.contentResolver.acquireContentProviderClient(ContactsContract.AUTHORITY)
+        } catch (e: SecurityException) {
+            logger.log(Level.WARNING, "Missing permissions for contacts authority", e)
+            null
+        }.use { provider ->
+            if (provider == null) {
+                logger.warning("Couldn't connect to content provider of contacts authority")
+                return // Don't continue without provider
+            }
+            val service = serviceRepository.getByAccountAndType(account.name, Service.TYPE_CARDDAV)
+                ?: return
+            val oldAddressBookAccounts = accountRepository.getAddressBookAccounts().filter {
+                account.name == accountManager.getUserData(it, "real_account_name")
+            }
```

Comment:
I'll add a comment: "Get all old address books = the ones which have a 'main account' in `real_account_name`."

(I already almost forgot how it worked previously 😄 So if we need to reconstruct what the migration did for instance when there are problems, more information is good.)